metrics/stackdriver_exporter.go

```diff
@@ -169,15 +189,10 @@ func getStackdriverSecret(sdconfig *stackdriverClientConfig) (*corev1.Secret, er
 		return nil, err
 	}
 
-	ns := sdconfig.GCPSecretNamespace
-	if ns == "" {
-		ns = secretNamespaceDefault
-	}
-
-	sec, secErr := kubeclient.CoreV1().Secrets(ns).Get(sdconfig.GCPSecretName, metav1.GetOptions{})
+	sec, secErr := kubeclient.CoreV1().Secrets(secretNamespace).Get(secretName, metav1.GetOptions{})
```

Comment:
should this do any secret format validation?

Reply:
I'm not sure there's much to be done here, since we can't really validate the key itself. There is format validation in that any error getting/using the Secret including missing fields will get logged.

Reply:
Does this perform the validation of the structure of the secret? 

https://github.com/knative/pkg/blob/d62d6b389417178900372cc232c622c4b9c91e93/metrics/stackdriver_exporter.go#L203

Reply:
Added a descriptive error message of the secretDataFieldKey is missing that will get passed back up and logged.